kotlin/src/main/kotlin/com/linecorp/armeria/server/kotlin/ServerBuilder.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.kotlin
+
+import com.linecorp.armeria.server.Server
+import com.linecorp.armeria.server.ServerBuilder
+
+/**
+ * Builds a new [Server] configured by the given [block].
+ */
+fun buildServer(block: ServerBuilder.() -> Unit): Server {
```

Comment:
I'm not sure this DSL is really useful for Kotlin users.
The users need to check the documentation to use this DSL because they don't know where to start building a server.

I thought of the following styles. Because the users start building server using `Server` class.
```kt
Server.apply {
  http(8080)
  service("/", { ctx, req -> ... }
  ...
}
```
However, as you know, it is impossible to add a Java static method to a Java class using the extension function syntax because of the limitation of the Kotlin compiler.
https://youtrack.jetbrains.com/issue/KT-11968

Reply:
Thanks, your concern makes sense. And I agree that your suggestion is better, because users can recognize DSLs naturally.
This is just of curiosity, if Armeria support DSLs using Kotlin's new feature, should users use at least the version which static extension introduced?